### PR TITLE
Add `Rename-PodeOADefinitionTag` Function

### PR DESCRIPTION
**Pull request by mdaneri**
**Description of the Change:**
This pull request introduces a new function, `Rename-PodeOADefinitionTagName`, which allows renaming an existing OpenAPI definition tag in Pode. This enhancement simplifies the management of OpenAPI definition tags and ensures a more streamlined configuration process.

**Key Changes:**
- Added `Rename-PodeOADefinitionTagName` function.
- The function renames a specified `Definition Tag` or the default tag if no tag is specified.
- Ensures that the new tag name does not already exist.
- Updates the default and selected definition tags if applicable.
- Throws an error if used within a `Select-PodeOADefinition` script block.
- Includes improvements to the way the default OpenAPI definition tag is managed.

**Examples:**
Renaming a specific OpenAPI definition tag:
```powershell
Rename-PodeOADefinitionTag -Tag 'v.3' -NewTag 'v.3.0.3'
```

Renaming the default OpenAPI definition tag:
```powershell
Rename-PodeOADefinitionTag -NewTag 'NewDefault'
```

**Notes:**
- The function will throw an error if the new tag name already exists.
- The function cannot be used inside a `Select-PodeOADefinition` script block.

**Pester Tests:**
Included Pester tests to validate the functionality:
- Renaming a specific OpenAPI definition tag.
- Renaming the default OpenAPI definition tag.
- Handling errors when the new tag name already exists.
- Handling errors when used inside a `Select-PodeOADefinition` script block.

**Documentation:**
For detailed usage and examples, refer to the documentation file included in the pull request.

### CODE REVIEW

File: src/Private/Context.ps1 [hunk 1]
```diff
@@ -936,6 +936,13 @@ function Set-PodeWebConfiguration {
...
             Enabled = [bool]$Configuration.Compression.Enable
         }
+        OpenApi          = @{
+            DefaultDefinitionTag = [string](Protect-PodeValue -Value $Configuration.OpenApi.DefaultDefinitionTag -Default 'default')
+        }
+    }
+
+    if ($Configuration.OpenApi -and $Configuration.OpenApi.ContainsKey('UsePodeYamlInternal')) {
+        $Context.Server.Web.OpenApi.UsePodeYamlInternal = $Configuration.OpenApi.UsePodeYamlInternal
     }
```
Comment (Badgerati) [inline src/Private/Context.ps1:946]: This block is almost identical to the block just beneath it 😄

Comment (mdaneri) [reply src/Private/Context.ps1:946]: it was wrong I was sure I removed it. But I was wrong

File: docs/Tutorials/Routes/Overview.md
Comment (Badgerati) [inline docs/Tutorials/Routes/Overview.md:1]: These changes look like they're for #1353?

Comment (mdaneri) [reply docs/Tutorials/Routes/Overview.md:1]: Some of them are duplicated in different places because I needed them 
what changes are you referring?

Comment (Badgerati) [reply docs/Tutorials/Routes/Overview.md:1]: From what I can tell, potentially all the docs changes in `docs/Tutorials/Routes/Overview.md`, as they all reference the `Get-PodeBody`, `Get-PodeParameter` etc. functions being created in #1353

Comment (mdaneri) [reply docs/Tutorials/Routes/Overview.md:1]: I think I committed to the wrong branch 
I'm going to fix it

File: docs/Tutorials/OpenAPI/Overview.md
Comment (Badgerati) [inline docs/Tutorials/OpenAPI/Overview.md:1]: I think this file might be being added back in by error, as a previous PR just merged deleted this and split it into several smaller pages?

File: docs/Tutorials/OpenAPI/6MultipleDefinitions.md [hunk 2]
```diff
@@ -56,6 +56,29 @@ The default `Definition Tag` is named "default". This can be changed using the `
...
+You can also rename the default `Definition Tag` without specifying the `Tag` parameter. This updates the default tag to the new name provided.
+
+```powershell
+Rename-PodeOADefinitionTagName -NewTag 'NewDefault'
+```
+
+In this example, the default definition tag is renamed to `'NewDefault'`.
+
+!!! note
+The `Rename-PodeOADefinitionTagName` function cannot be used inside a `Select-PodeOADefinition` `[Scriptblock]`. Attempting to do so will result in an error.
```
Comment (Badgerati) [inline docs/Tutorials/OpenAPI/6MultipleDefinitions.md:80]: For notes, tips, etc. the content has to be indented, ie:
```md
!!! note
    The `Rename-PodeOADefinitionTagName` function cannot be used inside a `Select-PodeOADefinition` `[Scriptblock]`. Attempting to do so will result in an error.
```

**Review by Badgerati — CHANGES_REQUESTED** [review]

File: docs/Tutorials/OpenAPI/Overview.md
Comment (mdaneri) [reply docs/Tutorials/OpenAPI/Overview.md:1]: yes because there was a change on the file.
This file can be deleted

File: src/Public/OpenApi.ps1 [hunk 3]
```diff
@@ -3433,6 +3425,77 @@ function Select-PodeOADefinition {
...
+
+Rename the default OpenAPI definition tag
+
+.NOTES
+This function will throw an error if:
+- It is used inside a Select-PodeOADefinition ScriptBlock.
+- The new tag name already exists.
+- The current tag name does not exist.
+#>
+function Rename-PodeOADefinitionTagName {
```
Comment (Badgerati) [inline src/Public/OpenApi.ps1:3459]: The function might be better named `Rename-PodeOADefinitionTag`, dropping the "Name", as it's "DefinitionTag", "Tag", "OADefinitionTag" everywhere.

Comment (mdaneri) [reply src/Public/OpenApi.ps1:3459]: Done

**Review by Badgerati — APPROVED** [review]

**Pull request merged**